### PR TITLE
Argument validation

### PR DESCRIPTION
**Pull request by Sutharsan**
To prevent invalid API requests, validation of required arguments is added to 'country by uuid' and 'city by uuid' requests.

### CODE REVIEW

**Comment by bartfeenstra** [comment]
The traits don't know about their usage, e.g. whether or not their functionality is required for the requests. As such, the exception messages and method names should not contain these assumptions. Instead they should say the values are invalid (and contain more specific information when available).

`// No assertion. Exception is thrown.`: no problem, but PHPUnit will automatically complain if a test method does not test anything (has no asserts, and does not expect exceptions, and sets no expectations).

`$this->sut = CityByUuid::create($this->productionRequestHandler);`
Don't use the production handler when simply testing validation. Instead, use the mocked handler in `RequestBaseTestBase::requestHandler`.

File: tests/Request/CityByUuidTest.php [hunk 1]
```diff
@@ -106,6 +106,27 @@ public function testExecuteRealRequest($form, $instanceof)
...
+    public function testExecuteInvalidRequest($uuid, $language_codes)
+    {
+        $this->sut = CityByUuid::create($this->requestHandler);
+
+        $this->sut
+          ->setUuid($uuid)
+          ->setLanguageCodes($language_codes)
+          ->execute();
+
+        $this->fail('Invalid arguments must throw an exception.');
```
Comment (bartfeenstra) [inline tests/Request/CityByUuidTest.php]: This is redundant if `@expectedException` is specified. PHPUnit will log a test failure automatically if no exception of the specified type is thrown during the test.

File: tests/Request/MultilingualTraitTest.php [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+     *   An array of ISO 639-1 alpha-2 language codes.
+     */
+    protected $languageCodes = [];
+
+    /**
+     * An invalid language codes.
+     *
+     * @var string
+     */
+    protected $invalidLanguageCodes;
```
Comment (bartfeenstra) [inline tests/Request/MultilingualTraitTest.php]: The property name and its docblock description are plural, yet the type hint indicates a single string value.

File: tests/Request/MultilingualTraitTest.php [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        $reflection = new \ReflectionClass($this->sut);
+        $reflection_property = $reflection->getProperty('languageCodes');
+        $reflection_property->setAccessible(true);
+        $reflection_property->setValue($this->sut, $this->invalidLanguageCodes);
+
+        $reflection_method = $reflection->getMethod('validateRequiredLanguageCodes');
+        $reflection_method->setAccessible(true);
+        $reflection_method->invoke($this->sut);
+
+        $this->fail('Invalid language codes must throw an exception.');
```
Comment (bartfeenstra) [inline tests/Request/MultilingualTraitTest.php]: Not necessary here either.

File: tests/Request/UuidTraitTest.php [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    /**
+     * @covers ::setUuid
+     *
+     * @expectedException \InvalidArgumentException
+     */
+    public function testSetInvalidUuid() {
+
+        $this->sut->setUuid($this->invalidUuid);
+
+        $this->fail('Invalid UUID must throw an exception.');
```
Comment (bartfeenstra) [inline tests/Request/UuidTraitTest.php]: Not necessary here either.

File: tests/Request/UuidTraitTest.php [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        $reflection = new \ReflectionClass($this->sut);
+        $reflection_property = $reflection->getProperty('uuid');
+        $reflection_property->setAccessible(true);
+        $reflection_property->setValue($this->sut, $this->invalidUuid);
+
+        $reflection_method = $reflection->getMethod('validateRequiredUuid');
+        $reflection_method->setAccessible(true);
+        $reflection_method->invoke($this->sut);
+
+        $this->fail('Invalid UUID must throw an exception.');
```
Comment (bartfeenstra) [inline tests/Request/UuidTraitTest.php]: Not necessary here either.

File: tests/Request/CityByUuidTest.php (hunk 1 above)
Comment (Sutharsan) [inline tests/Request/CityByUuidTest.php]: Now removed. Your comment above "but PHPUnit will automatically complain ..." got me confused.

**Pull request closed without merging**